src/jit/compiler.cpp

```diff
@@ -4194,11 +4194,17 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
     assert(!fgComputePredsDone);
     if (fgCheapPredsValid)
     {
-        // Remove cheap predecessors before inlining; allowing the cheap predecessor lists to be inserted
-        // with inlined blocks causes problems.
+        // Remove cheap predecessors before inlining and fat call transformation;
+        // allowing the cheap predecessor lists to be inserted causes problems
+        // with splitting existing blocks.
         fgRemovePreds();
     }
 
+    if (IsTargetAbi(CORINFO_CORERT_ABI) && doesMethodHaveFatPointer())
+    {
+        fgTransformFatCalli();
```

Comment:
I believe it could fairly easily be moved back after inlining since `fgInsertInlineeBlocks` will merge inlinee `optMethodFlags` into the root method. 

IIRC from discussions the "harder" part was when we were talking about deferring this even further, say to lower.